src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,134 @@
+        KalmanFilter<N1, N1, N1> kalman = new KalmanFilter<>(Nat.N1(), Nat.N1(), stateSpace,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(ENCODER_TOLERANCE),
+                Constants.ROBOT_TIMEOUT
+        );
+        LinearQuadraticRegulator<N1, N1, N1> lqr = new LinearQuadraticRegulator<>(stateSpace, VecBuilder.fill(VELOCITY_TOLERANCE),
+                VecBuilder.fill(Constants.VOLTAGE),
+                Constants.ROBOT_TIMEOUT // time between loops, DON'T CHANGE
+        );
+        this.stateSpacePredictor = new LinearSystemLoop<>(stateSpace, lqr, kalman, Constants.VOLTAGE, Constants.ROBOT_TIMEOUT);
+    }
+
+    /**
+     * @return the velocity of the shooter. [rps]
+     * @see #setVelocity(double)
+     */
+    public double getVelocity() {
+        return rpsUnitModel.toVelocity(main.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Set the velocity to apply by the motor.
+     *
+     * @param velocity the desired velocity at which the motor will rotate. [rps]
+     * @see #setPower(double)
+     */
+    public void setVelocity(double velocity) {
+        stateSpacePredictor.setNextR(VecBuilder.fill(velocity)); //r = reference
+        stateSpacePredictor.correct(VecBuilder.fill(getVelocity()));
+        stateSpacePredictor.predict(Constants.ROBOT_TIMEOUT); //every 20 ms
+
+        double voltageToApply = stateSpacePredictor.getU(0); // u = input, calculated by the input.
+        // returns the voltage to apply (between 0 and 12)
+        setPower(voltageToApply / Constants.VOLTAGE); // map to be between 0 and 1
+    }
+
+    /**
+     * Set the power to apply by the motor.
+     *
+     * @param power the power at which the motor will rotate. [percentage, between 0 and 1]
+     * @see #stop()
+     */
+    public void setPower(double power) {
+        main.set(TalonFXControlMode.PercentOutput, power, DemandType.ArbitraryFeedForward, Constants.Shooter.ARBITRARY_FEED_FORWARD);
+    }
+
+    /**
+     * Estimate the velocity that the shooter should apply in order to reach the target.
+     * The function created by running an experiment of different velocities for the "most used" distances.
+     * See the experiment results in (TODO: Add the path).
+     *
+     * @param distance the distance from the target.
+     * @return the velocity that should be applied by the shooter in order to reach the target.[rps]
+     */
+    public double estimateVelocity(double distance) {
+        return distance;
+    }
+
+    /**
+     * Get whether the flywheel has reached the desired velocity in order to reach the target.
+     * Also, this function checks whether the flywheel has enough velocity in order to move the motor in first place.
+     *
+     * @param desiredVelocity the desired velocity at the motor will rotate. [rps]
+     * @return whether the flywheel reaches the desired velocity.
+     */
+    public boolean isReady(double desiredVelocity) {
```

Comment:
"Setpoint" is the technical term:
```suggestion
    public boolean isReady(double setpoint) {
```
Or:
```suggestion
    public boolean isReady(double velocitySetpoint) {
```
I think the first is better.